File: mobile/src/api/stories.api.js
import { http } from './http';

export const fetchFeedStories = () => http.get('/stories/feed').then((r) => r.data);
export const fetchUserStories = (userId) => http.get(`/stories/user/${userId}`).then((r) => r.data);
export const createStory = (payload) => http.post('/stories', payload).then((r) => r.data);
export const viewStory = (storyId) => http.post(`/stories/${storyId}/view`).then((r) => r.data);
export const fetchStoryViewers = (storyId) => http.get(`/stories/${storyId}/viewers`).then((r) => r.data);
export const deleteStory = (storyId) => http.delete(`/stories/${storyId}`).then((r) => r.data);

File: mobile/lib/features/feed/data/stories_api.dart
import 'package:dio/dio.dart';
import 'package:flutter_riverpod/flutter_riverpod.dart';

import '../../../core/api_client.dart';
import '../../../models/story.dart';

/// Mirrors the RN app's `src/api/stories.api.js`.
class StoriesApi {
  final Dio _dio;

  StoriesApi(this._dio);

  Future<List<StoryGroup>> fetchStoriesFeed() async {
    final res = await _dio.get<List<dynamic>>('/stories/feed');
    return res.data!.map((e) => StoryGroup.fromJson(e as Map<String, dynamic>)).toList();
  }

  Future<List<Story>> fetchUserStories(String userId) async {
    final res = await _dio.get<List<dynamic>>('/stories/user/$userId');
    return res.data!.map((e) => Story.fromJson(e as Map<String, dynamic>)).toList();
  }

  Future<void> viewStory(String storyId) => _dio.post('/stories/$storyId/view');

  Future<List<StoryViewerEntry>> fetchStoryViewers(String storyId) async {
    final res = await _dio.get<List<dynamic>>('/stories/$storyId/viewers');
    return res.data!.map((e) => StoryViewerEntry.fromJson(e as Map<String, dynamic>)).toList();
  }

  /// `mediaUrl` comes from a prior `MediaApi.upload` call.
  Future<Story> createStory({required String mediaUrl, String mediaType = 'image', String? caption}) async {
    final res = await _dio.post<Map<String, dynamic>>('/stories', data: {
      'mediaUrl': mediaUrl,
      'mediaType': mediaType,
      'caption': ?(caption != null && caption.isNotEmpty ? caption : null),
    });
    return Story.fromJson(res.data!);
  }

  Future<void> deleteStory(String storyId) => _dio.delete('/stories/$storyId');
}

final storiesApiProvider = Provider<StoriesApi>((ref) => StoriesApi(ref.read(dioProvider)));
